adlfs/tests/test_user_agent.py

```diff
@@ -0,0 +1,85 @@
+import azure.storage.blob
+import pytest
+from azure.storage.blob.aio import BlobServiceClient as AIOBlobServiceClient
+
+from adlfs import AzureBlobFile, AzureBlobFileSystem
+from adlfs.tests.constants import CONN_STR, KEY
+from adlfs.utils import __version__ as __version__
+
+
+def assert_sets_adlfs_user_agent(mock_client_create_method):
+    mock_client_create_method.assert_called()
+    assert "user_agent" in mock_client_create_method.call_args.kwargs
```

Comment:
The `call_args` property returns only the last call to the mock. So, we may still have the problem where we are only asserting the call on the client created in the file system instead of the file object if the file object just reuses the client from the file system. Instead, we should probably:

1. Leverage [`call_args_list`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args_list) to ensure all calls made include the user agent. 
2. Add an `expected_call_count` with a default of `1` to this helper assertion so that for the file-related tests we can assert that the client was created twice; once in the filesystem and once in the file-like object as there is no way to avoid avoid creating the client in the file system object.